File: ts/strStr.ts
function strStr(haystack: string, needle: string): number {
  const splitted = haystack.split(needle);
  return splitted.length === 1 ? -1 : splitted[0].length;

  // return haystack.indexOf(needle);

  //   for (let i = 0; i < haystack.length; i++) {
  //     const letter = haystack[i];
  //     if (letter !== needle[0]) continue;
  //
  //     if (Array.from(needle).every((n, idx) => n === haystack[i + idx])) {
  //       return i;
  //     }
  //   }
  //
  //   return -1;
}

export default () => {
  const haystack = 'leetcode';
  const needle = 'leeto';

  const result = strStr(haystack, needle);
  console.log(result);
};

File: rs/src/str_str.rs
fn str_str(haystack: String, needle: String) -> i32 {
    let matched = haystack.match_indices(&needle).into_iter().nth(0);
    match matched {
        Some((usize, _)) => usize as i32,
        None => -1,
    }
}

pub fn main() {
    let haystack = "leetcode".to_string();
    let needle = "leeto".to_string();

    let result = str_str(haystack, needle);

    println!("{result}");
}
